client/components/data/query-site-guided-transfer/index.jsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingGuidedTransferStatus } from 'state/sites/guided-transfer/selectors';
+import { requestGuidedTransferStatus } from 'state/sites/guided-transfer/actions';
+
+class QuerySiteGuidedTransfer extends Component {
+
+	constructor( props ) {
+		super( props );
+		this.request = this.request.bind( this );
+	}
+
+	request( props = this.props ) {
+		if ( ! props.isRequesting && props.siteId ) {
+			props.requestGuidedTransferStatus( props.siteId );
+		}
+	}
+
+	componentWillMount() {
+		this.request();
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.isRequesting ||
+			! nextProps.siteId ||
+			( this.props.siteId === nextProps.siteId ) ) {
```

Comment:
since `this.request()` already checks for the existence of `props.siteId` and `! props.isRequesting` I think the logic in this `if` statement can be simplified to 
```js
if( this.props.siteId === nextProps.siteId ) {
    return;
}
```